contrib/node/src/python/pants/contrib/node/tasks/node_build.py

```diff
@@ -58,13 +58,23 @@ def _get_output_dir(self, target, node_installed_path):
       target.payload.output_dir if target.payload.build_script else ''))
 
   def execute(self):
+    targets = self.context.targets(predicate=self.is_node_module)
+    if not targets:
+      return
+
     node_paths = self.context.products.get_data(NodePaths)
-    runtime_classpath_product = self.context.products.get_data(
-      'runtime_classpath', init_func=ClasspathProducts.init_func(self.get_options().pants_workdir))
+    # This is required even if node does not need `compile_classpaths` because some downstream
+    # tasks requires `runtime_classpath` to be initialized correctly with `compile_classpaths`
+    compile_classpath = self.context.products.get_data('compile_classpath')
+    if compile_classpath is not None:
+      init_func = compile_classpath.copy
+    else:
+      init_func = ClasspathProducts.init_func(self.get_options().pants_workdir)
```

Comment:
If both of these calls used `ClasspathProducts.init_func`, you could avoid the conditional.

Reply:
They are separate: https://github.com/pantsbuild/pants/blob/14e0bd6b27910310e225ff9586c2410459153fd9/src/python/pants/backend/jvm/tasks/classpath_products.py#L162-L175

Reply:
Yes. But what I mean is: if you pass an init_func to the first call, then the result will never be None, and this if statement won't be necessary.